src/Engine/TranskribusClient.php

```diff
@@ -0,0 +1,150 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use App\Exception\OcrException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class TranskribusClient
+{
+
+    /** @var HttpClientInterface */
+    private $httpClient;
+
+    /** @var string transkribus access token. */
+    private $accessToken = '';
+
+    /** @var $config. */
+    private $config = [];
+
+    /** @var string Transcribed result. */
+    private const PROCESSES_URL = "https://transkribus.eu/processing/v1/processes";
+
+    /**
+     * TranskribusClient constructor.
+     * @param HttpClientInterface $httpClient
+     * @param string $accessToken
+     */
+    public function __construct(
+        HttpClientInterface $httpClient,
+        string $accessToken
+    ) {
+        $this->httpClient = $httpClient;
+        $this->accessToken = $accessToken;
+    }
+
+    /**
+     * @param string $imageURL
+     * @param mixed[] $config
+     * @return int
+     */
+    public function initProcess(string $imageURL, array $config = []): int
+    {
+        if ([] === $config) {
+            $config = [
+                'textRecognition' => [
+                    'htrId' => 38230,
+                ],
+            ];
+        }
+
+        $jsonBody = [
+            'config' => $config,
+            'image' => [
+                'imageUrl' => $imageURL,
+            ],
+        ];
+
+        $processId = 0;
+        $content = $this->request('POST', self::PROCESSES_URL, $jsonBody);
+        $parsedContent = (array) $content;
+        if (!empty($parsedContent)) {
+            if ($content->{'status'} === 'CREATED') {
+                $processId = $content->{'processId'};
+            }
+        }
+
+        return $processId;
+    }
+
+    /**
+     * @param int $processId
+     * @return string
+     */
+    public function retrieveProcessResult(int $processId): string
+    {
+        $url = self::PROCESSES_URL.'/'.$processId;
+
+        $content = $this->request('GET', $url, []);
+        $parsedContent = (array) $content;
+        $textResult = '';
+
+        if (!empty($parsedContent)) {
+            if ($content->{'status'} === 'FINISHED') {
+                $textContent = $content->{'content'};
+                $textResult = $textContent->{'text'};
+            }
+        }
+
+        return $textResult;
+    }
+
+    /**
+     * @param int $statusCode
+     * @return string
+     */
+    private function getErrorMessage(int $statusCode): string
+    {
+        $errorMessage = '';
+        switch ($statusCode) {
+            case 0:
+                $errorMessage = 'Transkribus API returned null';
```

Comment:
Should we be changing this message to something like `Could not parse result from Transkribus API` instead?